keyboards/planck/keymaps/orthodeluxe/keymap.c

```diff
@@ -0,0 +1,602 @@
+ */
+[_NAVLAYER] = LAYOUT_planck_grid(
+    _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,    XXXXXXX, KC_PGUP, HOMEKEY,  KC_UP,   ENDKEY,   PRTSCR,    LOCKSCREEN,
+    _______, KC_F5,   KC_F6,   KC_F7,   KC_F8,    XXXXXXX, KC_PGDN, KC_LEFT,  KC_DOWN, KC_RGHT,  APPSWITCH, _______,
+    _______, KC_F9,   KC_F10,  KC_F11,  KC_F12,   XXXXXXX, XXXXXXX, PREVDESK, KC_DOWN, NEXTDESK, TASKVIEW,  _______,
+    _______, _______, _______, _______, SYMLAYER, _______, _______, _______,  _______, _______,  _______,   _______
+),
+
+/* Utilities layer:
+ * ,-----------------------------------------------------------------------------------.
+ * |      |      |  Win |      | Reset|      |      |      |  Ins |      | Pause|      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |ScrlLk| Debug|      |      |      |      |      |      |      |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |CapsLk|      |      | NumLk|  Mac |      |      |      |      |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * |      |      |      |      |(hold)|    (hold)   |(hold)|      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_UTILSLAYER] = LAYOUT_planck_grid(
+    _______, XXXXXXX, WINDOWS, XXXXXXX, RESET,   XXXXXXX, XXXXXXX, XXXXXXX, KC_INS,  XXXXXXX, KC_PAUS, _______,
+    _______, XXXXXXX, KC_SLCK, DEBUG,   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, _______,
+    _______, XXXXXXX, XXXXXXX, KC_CAPS, XXXXXXX, XXXXXXX, KC_NLCK, MACOS,   XXXXXXX, XXXXXXX, XXXXXXX, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+)
+};
+
+#ifdef AUDIO_ENABLE
+  float plover_song[][2]     = SONG(PLOVER_SOUND);
+  float plover_gb_song[][2]  = SONG(PLOVER_GOODBYE_SOUND);
+#endif
+
+bool macos_mode = true;
+bool app_switch_active = false;
+static uint16_t timer;
+static uint16_t duration = 0;
+static bool double_tap = false;
+uint8_t active_modifiers = 0;
+bool shift_modifier_active = false;
+
+/* main loop */
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+
+    /* Store modifiers in booleans for easy access in underlying methods. */
+    active_modifiers = get_mods();
+    shift_modifier_active = active_modifiers & ((1<<((KC_LSFT) & 0x07)) | (1<<((KC_RSFT) & 0x07)));
+
+    /* Switch off timer when double click time is exceeded. */
+    if (timer > 0) {
+        duration = timer_elapsed(timer);
+        if (duration > TAPPING_TERM) { timer = 0; }
+    }
+        
+    /* Keep app_switch_active true, while a valid key is pressed. */
+    if (app_switch_active) {
+        if (keycode == NAVLAYER && record->event.pressed == false) {
+            unregister_code(macos_mode ? KC_LGUI : KC_LALT);
```

Comment:
You could actually use `keycode_config` here. 
```suggestion
            unregister_code(keycode_config(KC_LALT));
```

This use the bootmagic config to handle this. 
https://github.com/qmk/qmk_firmware/blob/59832e7654d5b016d46b08d302788fcd330a0d67/quantum/keycode_config.c#L26

Reply:
@drashna : you are right that I could use bootmagic here to swap KC_LALT and KC_LGUI. But I use the boolean macos_mode for many more functions in my code (see handle_lock_screen, handle_print_screen, type_eurosign, handle_home_key, handle_end_key, handle_show_task_view, handle_switch_desktop). I don't think bootmagic can be used there.

Reply:
Nope, but you can check they keymap_config structure for the bootmagic settings. 

https://github.com/qmk/qmk_firmware/blob/d21730774752b8bc439564b1e3b687ad2aa0922c/quantum/keycode_config.h#L30-L42

And you can call those settings by using `keymap_config.swap_lalt_lgui` for instance.  

This may not be perfect for what you need, but it may be close. 

Also, you can create your own that is persistent, actually: 
https://docs.qmk.fm/#/custom_quantum_functions?id=persistent-configuration-eeprom

Reply:
@drashna : Thanks for your comments. I learned new stuff and I added Bootmagic Lite. Your comments helped me to greatly simplify the code.

Reply:
You are very welcome!